File: rust/src/models/slot_fee_structure.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{
    slot_downgrade_policy_enum::SlotDowngradePolicyEnum,
    slot_upgrade_policy_enum::SlotUpgradePolicyEnum,
};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct SlotFeeStructure {
    pub downgrade_policy: SlotDowngradePolicyEnum,

    pub slot_unit_name: String,

    pub upgrade_policy: SlotUpgradePolicyEnum,
}

impl SlotFeeStructure {
    pub fn new(
        downgrade_policy: SlotDowngradePolicyEnum,
        slot_unit_name: String,
        upgrade_policy: SlotUpgradePolicyEnum,
    ) -> Self {
        Self {
            downgrade_policy,
            slot_unit_name,
            upgrade_policy,
        }
    }
}

File: typescript/src/models/slotFeeStructure.ts
// this file is @generated
import {
  type SlotDowngradePolicyEnum,
  SlotDowngradePolicyEnumSerializer,
} from "./slotDowngradePolicyEnum";
import {
  type SlotUpgradePolicyEnum,
  SlotUpgradePolicyEnumSerializer,
} from "./slotUpgradePolicyEnum";

export interface SlotFeeStructure {
  downgradePolicy: SlotDowngradePolicyEnum;

  slotUnitName: string;

  upgradePolicy: SlotUpgradePolicyEnum;
}

export const SlotFeeStructureSerializer = {
  _fromJsonObject(object: any): SlotFeeStructure {
    return {
      downgradePolicy: SlotDowngradePolicyEnumSerializer._fromJsonObject(
        object["downgrade_policy"]
      ),
      slotUnitName: object["slot_unit_name"],
      upgradePolicy: SlotUpgradePolicyEnumSerializer._fromJsonObject(
        object["upgrade_policy"]
      ),
    };
  },

  _toJsonObject(self: SlotFeeStructure): any {
    return {
      downgrade_policy: SlotDowngradePolicyEnumSerializer._toJsonObject(
        self.downgradePolicy
      ),
      slot_unit_name: self.slotUnitName,
      upgrade_policy: SlotUpgradePolicyEnumSerializer._toJsonObject(self.upgradePolicy),
    };
  },
};
